packages/cache/src/cache.ts

```diff
@@ -185,14 +175,32 @@ export async function saveCache(
     const fileSizeLimit = 10 * 1024 * 1024 * 1024 // 10GB per repo limit
     const archiveFileSize = utils.getArchiveFileSizeInBytes(archivePath)
     core.debug(`File Size: ${archiveFileSize}`)
-    if (archiveFileSize > fileSizeLimit) {
+    if (archiveFileSize > fileSizeLimit && !utils.isGhes()) {
```

Comment:
Add a comment here as to why we are excluding this check for GHES

Reply:
Removed this code altogether

Reply:
Brought back this code and added a comment.